test/forms/behaviours/DateBehaviours.scala

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.behaviours
+
+import cats.implicits.catsSyntaxOptionId
+import forms.standard.StandardErrorKeys._
+import org.scalacheck.Gen
+import org.scalatest.OptionValues
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import play.api.data.{Form, FormError}
+import utils.generators.Generators
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+trait DateBehaviours extends AnyFreeSpec with ScalaCheckPropertyChecks with Generators with Matchers with OptionValues {
```

Comment:
Just an observation (no action needed)

If the trait is only used by one Spec, maybe we should put it directly into this Spec. 

Also the long chain of mixins here will be make this trait hard to re-use later. Not to mention it extends AnyFreeSpec which enforces particular test style. I would prefer to have in all tests the same AnyWordSpecLike style if possible.